src/aks-preview/azext_aks_preview/_help.py

```diff
@@ -46,6 +46,10 @@
         - name: --node-osdisk-size
           type: int
           short-summary: Size in GB of the OS disk for each node in the node pool. Minimum 30 GB.
+        - name: --node-osdisk-type
+          type: string
+          short-summary: OS disk type to be used for machines in a given agent pool. Allowed values are 'Ephemeral'
```

Comment:
Since `arg_type=get_enum_type([CONST_OS_DISK_TYPE_MANAGED, CONST_OS_DISK_TYPE_EPHEMERAL])` is used in _param.py, the framework will automatically add "Allowed values" list for it. We do not need to add manually here.

And if the default value is "Managed", we can directly define the custom function as below:
> def aks_create(cmd,     # pylint: disable=too-many-locals,too-many-statements,too-many-branches
               client,	               client,
               resource_group_name,	               resource_group_name,
               name,	               name,
               ssh_key_value,	               ssh_key_value,
               dns_name_prefix=None,	               dns_name_prefix=None,
               location=None,	               location=None,
               admin_username="azureuser",	               admin_username="azureuser",
               windows_admin_username=None,	               windows_admin_username=None,
               windows_admin_password=None,	               windows_admin_password=None,
               kubernetes_version='',	               kubernetes_version='',
               node_vm_size="Standard_DS2_v2",	               node_vm_size="Standard_DS2_v2",
               node_osdisk_type="**Managed**",

The framework will also add Default value in help message automatically.

Reply:
I'll remove it from the help 👍 but I don't think we want to default this value client side, we want it to be empty and we will default it server side. 

In October, the defaulting behavior on server side will get slightly more complex when the value isn't provided and I would strongly prefer not to copy that defaulting behavior into the client.